### PR TITLE
Add support for admin ServerUpdate to update containers simultaneously

### PR DESCRIPTION
**Pull request by harshavardhana**

### CODE REVIEW

**Comment by harshavardhana** [comment]
This PR testing would depend on the PR https://github.com/minio/minio/pull/10107 here

File: pkg/controller/cluster/main-controller.go [hunk 1]
```diff
@@ -516,8 +516,82 @@ func (c *Controller) syncHandler(key string) error {
...
 		}
+
 		// keep track of all replicas
 		totalReplicas += ss.Status.Replicas
+		images = append(images, ss.Spec.Template.Spec.Containers[0].Image)
+	}
+
+	var sameImageCount int
+	for _, image := range images {
+		if mi.Spec.Image == image {
```
Comment (nitisht) [inline pkg/controller/cluster/main-controller.go:527]: This condition will always fail when there is a new image update requested, sent a PR https://github.com/harshavardhana/minio-operator/pull/1 PTAL

File: pkg/apis/minio.min.io/v1/constants.go [hunk 2]
```diff
@@ -72,11 +73,15 @@ const MinIOVolumeSubPath = ""
 // DefaultMinIOImage specifies the default MinIO Docker hub image
 const DefaultMinIOImage = "minio/minio:RELEASE.2020-07-22T00-26-33Z"
 
+// DefaultMinIOUpdateURL specifies the default MinIO URL where binaries are
+// pulled from during MinIO upgrades
+const DefaultMinIOUpdateURL = "https://dl.min.io/server/minio/release/" + runtime.GOOS + "-" + runtime.GOARCH + "/archive/"
+
 // DefaultMinIOAccessKey specifies default access key for Tenant
-const DefaultMinIOAccessKey = "AKIAIOSFODNN7EXAMPLE"
+const DefaultMinIOAccessKey = "ACCESS-KEY-EXAMPLE"
```
Comment (kerneltime) [inline pkg/apis/minio.min.io/v1/constants.go:81]: Nit: We should not have a default password that scanners try to discovery lazy ops..

Comment (harshavardhana) [reply pkg/apis/minio.min.io/v1/constants.go:81]: yeah that's why renaming it as `-EXAMPLE` - not sure if we don't want a default perhaps fail if its not set?

Comment (kerneltime) [reply pkg/apis/minio.min.io/v1/constants.go:81]: Yeah the operator can fail the instance creation if secret not set.

Comment (harshavardhana) [reply pkg/apis/minio.min.io/v1/constants.go:81]: Looks like its not used, removing it.

**Comment by harshavardhana** [comment]
Tested all the changes are working as expected 

```diff
diff --git a/examples/tenant.yaml b/examples/tenant.yaml
index 0abf379..254f440 100644
--- a/examples/tenant.yaml
+++ b/examples/tenant.yaml
@@ -42,7 +42,8 @@ spec:
       prometheus.io/port: "9000"
       prometheus.io/scrape: "true"
   ## Registry location and Tag to download MinIO Server image
-  image: minio/minio:RELEASE.2020-07-22T00-26-33Z
+  image: y4m4/minio:RELEASE.2020-07-21T01-08-20Z
+  #image: minio/minio:RELEASE.2020-07-22T00-26-33Z
   ## A ClusterIP Service will be created with the given name
   serviceName: minio-internal-service
   zones:
@@ -56,7 +57,8 @@ spec:
             - ReadWriteOnce
           resources:
             requests:
-              storage: 1Ti
+              storage: 20Gi
+          storageClassName: direct.csi.min.io
   ## Mount path where PV will be mounted inside container(s). Defaults to "/export".
   mountPath: /export
   ## Sub path inside Mount path where MinIO starts. Defaults to "".
diff --git a/minio-operator.yaml b/minio-operator.yaml
index 58617d6..7dd915c 100644
--- a/minio-operator.yaml
+++ b/minio-operator.yaml
@@ -276,7 +276,7 @@ spec:
         name: minio-operator
     spec:
       containers:
-        - image: minio/k8s-operator:3.0.0
-          imagePullPolicy: IfNotPresent
+        - image: y4m4/k8s-operator:test
+          imagePullPolicy: Always
           name: minio-operator
       serviceAccountName: minio-operator
```

Apply the following diff locally and then change the container image after deployment 
with a second `kubectl apply -f` you can see the container restart and update first 
before they do rolling upgrades.

**Pull request merged**